src/crimson/osd/ops_executer.cc

```diff
@@ -434,101 +436,122 @@ OpsExecuter::execute_op(OSDOp& osd_op)
   case CEPH_OSD_OP_SYNC_READ:
     [[fallthrough]];
   case CEPH_OSD_OP_READ:
-    return do_read_op([&osd_op] (auto& backend, const auto& os) {
+    return (OpsExecuter::osd_op_errorator::future<>) do_read_op(
```

Comment:
I'm not following why you need to cast to future<> here.

Reply:
Sorry this change should be dropped, it was added when I was working on another way of implementation

Reply:
Sorry, this change should be dropped, it was added when I was thinking another way of implementation